src/global/project/manifest.rs

```diff
@@ -472,19 +472,25 @@ impl FromStr for Mapping {
 pub enum ExposedType {
     #[default]
     All,
-    Subset(Vec<Mapping>),
+    Filter(Vec<PackageName>),
+    Mappings(Vec<Mapping>),
 }
 
 impl ExposedType {
-    pub fn from_mappings(mappings: Vec<Mapping>) -> Self {
-        match mappings.is_empty() {
-            true => Self::All,
-            false => Self::Subset(mappings),
+    pub fn new(mappings: Vec<Mapping>, filter: Vec<PackageName>) -> Self {
```

Comment:
just to not forget - I think we will need to rethink a little new method here, because if I pass both of them I will still use only mappings